### PR TITLE
Update Azure SDK

### PR DESCRIPTION
**Pull request by paulmey**
Changed to using context for cancellation and waiting for futures where applicable.
Added acceptance tests
Snapped to specific api-versions per service

### CODE REVIEW

**Comment by mwhooker** [comment]
This looks good to me. The tests are failing because the code needs to be formatted (`make format`).

I had a thought when I saw all the context.TODO()s in https://github.com/hashicorp/packer/pull/6224/files#diff-168d8353ae7a59ff1b03a1ec9bab936a

The runner itself doesn't get a context because it's up to the builder to start the runner, and so we have limited access to that as a framework.

What you could do (totally optional), is create a background context at the start of `Run`, save it in the Builder struct, and then cancel it in `Cancel`. This isn't optimal since go really doesn't want you to do this (I think it interferes with static analysis), but should be a better alternative than completely changing the Builder interface. If you did this, then cancelling the azure build while the `Run` preamble is running would immediately end, instead of waiting for it to try to run the first step.

**Comment by paulmey** [comment]
Thanks @mwhooker, I liked that idea of having a context for the non-multistep parts of `Builder.Run`. PTAL and see if this is what you had in mind.

File: builder/azure/arm/step_capture_image.go [hunk 1]
```diff
@@ -86,25 +83,24 @@ func (s *StepCaptureImage) Run(_ context.Context, state multistep.StateBag) mult
...
+		if isManagedImage {
+			s.say(fmt.Sprintf(" -> Image ResourceGroupName   : '%s'", targetManagedImageResourceGroupName))
+			s.say(fmt.Sprintf(" -> Image Name                : '%s'", targetManagedImageName))
+			s.say(fmt.Sprintf(" -> Image Location            : '%s'", targetManagedImageLocation))
+			err = s.captureManagedImage(ctx, targetManagedImageResourceGroupName, targetManagedImageName, imageParameters)
+		} else {
+			err = s.captureVhd(ctx, resourceGroupName, computeName, vmCaptureParameters)
+		}
+	}
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (paulmey) [reply builder/azure/arm/step_capture_image.go]: the block above is trying to fail some more and potentially sets `err`... so I'll leave it as is.

File: builder/azure/arm/step_capture_image.go [hunk 2]
```diff
@@ -86,25 +83,24 @@ func (s *StepCaptureImage) Run(_ context.Context, state multistep.StateBag) mult
 	s.say(fmt.Sprintf(" -> Compute Name              : '%s'", computeName))
 	s.say(fmt.Sprintf(" -> Compute Location          : '%s'", location))
 
-	result := common.StartInterruptibleTask(
```
[the remark this answers is not in the document]
Comment (paulmey) [reply builder/azure/arm/step_capture_image.go]: Done.

File: builder/azure/arm/step_create_resource_group_test.go [hunk 3]
```diff
@@ -195,7 +201,7 @@ func createTestStateBagStepCreateResourceGroup() multistep.StateBag {
 		"tag01": &value,
 	}
 
-	stateBag.Put(constants.ArmTags, &tags)
```
[the remark this answers is not in the document]
Comment (paulmey) [reply builder/azure/arm/step_create_resource_group_test.go]: Checked, it should be good...

**Pull request merged**